link/src/main/java/com/stripe/android/link/ui/wallet/WalletViewModel.kt

```diff
@@ -149,20 +150,53 @@ internal class WalletViewModel @Inject constructor(
                 }
             )
         } else {
-            val params = createConfirmStripeIntentParams(
+            val params = createPaymentMethodCreateParams(
                 selectedPaymentDetails = selectedPaymentDetails,
-                linkAccount = linkAccount
+                linkAccount = linkAccount,
             )
 
-            confirmationManager.confirmStripeIntent(params) { result ->
-                result.fold(
-                    onSuccess = ::handleConfirmPaymentSuccess,
-                    onFailure = ::onError
-                )
+            val nextStep = intentConfirmationInterceptor.intercept(
```

Comment:
Is there a way to share the code between PaymentMethodViewModel and WalletViewModel? Seems like we are repeating the same code here.